core/src/HIP/Kokkos_HIP_ReduceScan.hpp

```diff
@@ -477,22 +477,24 @@ __device__ bool hip_single_inter_block_reduce_scan(
 }
 
 // Size in bytes required for inter block reduce or scan
-template <bool DoScan, class FunctorType, class ArgTag>
+template <bool DoScan, class ArgTag, class ValueType, class FunctorType>
```

Comment:
Why did you change the other of the template parameters?  You didn't change it in the CUDA backend.

Reply:
So I can take advantage of template argument deduction. I can mirror for Cuda if you prefer.

Reply:
Yes, please.